### PR TITLE
Update DraftJS Mention Plugin.

### PR DESCRIPTION
**Pull request by igorschoester**
## Summary

This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* Updated `draft-js-mention-plugin` to get rid of the console warning.
*  Refactored `serialize` and `unserialize`. Partly for the major version upgrade. And partly for the marking issue.
* Now passes the tags along to ignore non-tag affix matches.

## Test instructions

This PR can be tested by following these steps:

* Run the tests.
* Check in the stand-alone.
* Link with the plugin and test.

Fixes https://github.com/Yoast/wordpress-seo/issues/9684 and https://github.com/Yoast/wordpress-seo/issues/9653

### CODE REVIEW

File: composites/Plugin/SnippetEditor/serialization.js [hunk 1]
```diff
@@ -1,14 +1,18 @@
-const ENTITY_FORMAT = /%%([a-zA-Z_]+)%%/;
+import forEach from "lodash/forEach";
+import sortedIndexBy from "lodash/sortedIndexBy";
+import trim from "lodash/trim";
+
+const AFFIX = "%%";
```
Comment (Xyfi) [inline composites/Plugin/SnippetEditor/serialization.js]: Could be renamed to `CIRCUMFIX`.

**Comment by Xyfi** [comment]
CR :+1:

File: composites/Plugin/SnippetEditor/serialization.js [hunk 2]
```diff
@@ -64,69 +84,105 @@ export function serializeEditor( rawContent ) {
...
 
+/**
+ * Find all indices of a search term in a string.
+ *
+ * @param {string} searchTerm The term to search for.
+ * @param {string} text       The text to search in.
+ *
+ * @returns {Array} Array of found indices.
+ */
+const getIndicesOf = ( searchTerm, text ) => {
```
Comment (boblinthorst) [inline composites/Plugin/SnippetEditor/serialization.js]: We probably want to unit test this function in the future.

File: composites/Plugin/SnippetEditor/serialization.js [hunk 3]
```diff
@@ -64,69 +84,105 @@ export function serializeEditor( rawContent ) {
...
 /**
  * Unserializes a piece of content into DraftJS data.
  *
  * @param {string} content The content to unserialize.
+ * @param {Array} tags The tags for the DraftJS mention plugin.
+ *
  * @returns {Object} The raw data ready for convertFromRaw.
  */
-export function unserializeEditor( content ) {
+export function unserializeEditor( content, tags ) {
```
Comment (boblinthorst) [inline composites/Plugin/SnippetEditor/serialization.js]: As discussed in person. In the future, we're gonna want to extract some parts of this function to make ik more comprehensive.

**Pull request merged**